### PR TITLE
Pylint adjustments

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added
- [x] The relevant author information has been added to `AUTHORS.rst` and `.zenodo.json`

### What kind of change does this PR introduce?

* Addresses the pylint stack overflow crashes that have been happening since the release of pylint v2.15
* Moves the `pylint` configuration to its own `.pylintrc` based on the examples offered at pylint
* Updates makefile to use `.pylintrc`
* Removed pylint from tox (issues with virtualenv and pylint)
* Addresses some easily-addressed pylint warnings

### Does this PR introduce a breaking change?

No.

### Other information:

I've removed the Codefactor.io check, as the configuration for it was invalid this entire time (hence why so few errors were ever raised). `pylint` is much more synchronized with our code style, and we should aim to address/mark-for-exception the issues that we deem to be errors. Once that work is done, we can remove the `--exit-zero` flag and allow it to fail builds.

### CODE REVIEW

File: xclim/core/indicator.py [hunk 1]
```diff
@@ -1684,6 +1686,7 @@ def _merge_attrs(dbase, dextra, attr, sep):
                 data, identifier=identifier, module=module_name
             )
 
+        # FIXME: Handle the potential exceptions explicitly
```
Comment (aulemahal) [inline xclim/core/indicator.py:1689]: Well I know this is what pylint whines about, but the point of this construct is that we actually don't know what could cause the function to fail. BUT, we don't want to stop the execution of the program if it does.
I don't think the construction makes any sense if we narrow it down to a list of exceptions.

File: xclim/sdba/adjustment.py [hunk 2]
```diff
@@ -134,10 +134,10 @@ def _harmonize_units(cls, *inputs, target: str | None = None):
         return (convert_units_to(inda, target) for inda in inputs), target
 
     @classmethod
-    def _train(ref, hist, **kwargs):
+    def _train(cls, hist, **kwargs):
```
Comment (aulemahal) [inline xclim/sdba/adjustment.py:137]:
```suggestion
    def _train(cls, ref, hist, **kwargs):
```
My bad for this signature. The goal was to show what it should look like.

File: xclim/sdba/adjustment.py [hunk 3]
```diff
@@ -134,10 +134,10 @@ def _harmonize_units(cls, *inputs, target: str | None = None):
...
         raise NotImplementedError()
 
-    def _adjust(sim, *args, **kwargs):
+    def _adjust(self, *args, **kwargs):
```
Comment (aulemahal) [inline xclim/sdba/adjustment.py:140]:
```suggestion
    def _adjust(self, sim, *args, **kwargs):
```

**Review by aulemahal — APPROVED** [review]
Good with me!

File: xclim/core/indicator.py (hunk 1 above)
Comment (Zeitsperre) [reply xclim/core/indicator.py:1689]: Understood, I don't think there's any issue with marking "exceptions" to the pylint rules, so long as we're explicit about them.

**Pull request merged**